libs/editor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java

```diff
@@ -1635,4 +1656,71 @@ public void onConnectionStatusChange(boolean isConnected) {
             mEditorFragmentListener.onMediaRetryAll(mFailedMediaIds);
         }
     }
+
+    @Override
+    public boolean canIntercept(@NonNull WebResourceRequest request) {
+        Uri url = request.getUrl();
+        String siteURL = (String) (mSettings != null ? mSettings.get("siteURL") : "");
+        String siteHostedMedia = siteURL + "/.*\\.(jpg|jpeg|png|gif|bmp|webp|mp4|mov|avi|mkv|mp3|wav|flac)(\\?.*)?$";
+        Pattern pattern = Pattern.compile(siteHostedMedia, Pattern.CASE_INSENSITIVE);
```

Comment:
I'm not sure a regex is the way to do this? URL parsing is probably more performant and accurate. WDYT?

Reply:
Makes sense to me. I was merely unfamiliar with the built-in URL parsing. I'll refactor this.

Reply:
Refactored in a9fe1d56d3b4b7c5796337f27c463095400a5ddb.